### ISSUE TITLE
read_csv throws Illegal instruction (core dumped) on a simple csv file

### ISSUE THREAD

**Issue by mknorps** [pandas-dev/pandas#18891]
#### Code Sample, a copy-pastable example if possible

```python
import numpy as np
import pandas as pd
from os.path import expanduser

datadir =  expanduser("~") + "/Documents/"
ftest = datadir + "test.csv"

def read_f(f):
    print ("read_files",f)
    df = pd.read_csv(f)

if __name__=="__main__":
    read_f(ftest)
```

test.csv:
a,b,c
1,2,3
4,5,6

#### Problem description

When calling "pd.read_csv" a core dump error occurs.
Program output:

```
mknorps@machine:~/Documents$ python project.py
Run of cost estimation
read_files /home/knorpma1/Documents/cost_estimation/data/test.csv
Illegal instruction (core dumped)
```

I cannot load any CSV file with pandas read_csv.
When I debug (pdb) the same error occurs and the debugger gets a core dump error.

#### setup
Python 3.6.3 :: Anaconda custom (32-bit)
mknorps@machine:~/Documents/project$ conda list
# packages in environment at /home/mknorps/anaconda3:
#
_ipyw_jlab_nb_ext_conf    0.1.0            py36h1427579_0  
alabaster                 0.7.10           py36hc729686_0  
anaconda                  custom           py36hc37a2b5_0  
anaconda-client           1.6.6            py36h40bb92b_0  
anaconda-navigator        1.6.10           py36hfcb3893_0  
anaconda-project          0.8.2            py36h453df49_0  
asn1crypto                0.23.0           py36h58c7482_0  
astroid                   1.5.3            py36h4acdba0_0  
astropy                   2.0.2            py36h2cc39e1_3  
attrs                     17.3.0           py36h3cf7522_0  
babel                     2.5.0            py36h0f939e7_0  
backports                 1.0              py36h9935294_1  
backports.shutil_get_terminal_size 1.0.0            py36h53f91a9_2  
beautifulsoup4            4.6.0            py36h6ea3382_0  
bitarray                  0.8.1            py36ha18d170_0  
bkcharts                  0.2              py36hf4da8f5_0  
blaze                     0.11.3           py36hfc1e9e3_0  
bleach                    2.1.1            py36h0d22a17_0  
bokeh                     0.12.13          py36h2f9c1c0_0  
boto                      2.48.0           py36heef2262_1  
bottleneck                1.2.1            py36h4dbbece_0  
bzip2                     1.0.6                h1033540_2  
ca-certificates           2017.08.26           h1d4fec5_0  
cairo                     1.14.10              h3f859d5_6  
certifi                   2017.11.5        py36h07e792d_0  
cffi                      1.11.2           py36hcc0a9a0_0  
chardet                   3.0.4            py36h5b25bb0_1  
click                     6.7              py36h8d67d36_0  
cloudpickle               0.5.2            py36hff59154_0  
clyent                    1.2.2            py36hdef67e0_0  
colorama                  0.3.9            py36hbfa1332_0  
conda                     4.3.31                   py36_0  
conda-build               3.1.6                    py36_0  
conda-env                 2.6.0                h36134e3_0  
conda-verify              2.0.0            py36h47aa4cd_0  
contextlib2               0.5.5            py36h9fd5caf_0  
coverage                  4.4.2            py36h230b894_0  
cryptography              2.1.4            py36h509455f_0  
curl                      7.55.1               h04b7030_4  
cycler                    0.10.0           py36hb8660b5_0  
cython                    0.27.3           py36h8b1c79f_0  
cytoolz                   0.8.2            py36h1899b5f_0  
dask                      0.16.0           py36h0571a84_0  
dask-core                 0.16.0           py36h0272949_0  
datashape                 0.5.4            py36h93bcdfc_0  
dbus                      1.10.22              ha863c99_0  
decorator                 4.1.2            py36h1a5868c_0  
distributed               1.20.2                   py36_0  
docutils                  0.14             py36hb988c27_0  
entrypoints               0.2.3            py36he4c36a9_2  
et_xmlfile                1.0.1            py36h7e62236_0  
expat                     2.2.5                h8d48cb0_0  
fastcache                 1.0.2            py36hd3723bf_0  
filelock                  2.0.13           py36hb5244f6_0  
flask                     0.12.2           py36h8077ff0_0  
flask-cors                3.0.3            py36ha5d92f3_0  
fontconfig                2.12.4               hcb742b8_1  
freetype                  2.8                  hb4d7b56_1  
get_terminal_size         1.0.0                hb836b69_0  
gevent                    1.2.2            py36hcd00a01_0  
glib                      2.53.6               hd466c72_1  
glob2                     0.6              py36h18154a7_0  
gmp                       6.1.2                h8afb2fd_1  
gmpy2                     2.0.8            py36h32a06e3_2  
graphite2                 1.3.10               h8f21b5a_0  
greenlet                  0.4.12           py36h3518183_0  
gst-plugins-base          1.12.2               h3b3eec3_0  
gstreamer                 1.12.2               ha52c2f3_0  
h5py                      2.7.0               np113py36_0  
harfbuzz                  1.5.0                hc3e4629_0  
hdf5                      1.8.17                        2  
heapdict                  1.0.0            py36h1a430f0_0  
html5lib                  1.0.1            py36h2f9c1c0_0  
hypothesis                3.38.5           py36h250ea84_0  
icu                       58.2                 he175447_1  
idna                      2.6              py36ha4b94b8_1  
imageio                   2.2.0            py36h6e635ad_0  
imagesize                 0.7.1            py36he490439_0  
intel-openmp              2018.0.0             h172fb5a_8  
ipykernel                 4.7.0            py36h2f9c1c0_0  
ipython                   6.2.1            py36h4218bc2_1  
ipython_genutils          0.2.0            py36h6759a34_0  
ipywidgets                7.0.5            py36h5d44178_0  
isort                     4.2.15           py36h0bc576f_0  
itsdangerous              0.24             py36h69a58be_1  
jbig                      2.1                  hfc6e72a_0  
jdcal                     1.3              py36h5c236aa_0  
jedi                      0.11.0                   py36_2  
jinja2                    2.10             py36hbf2117e_0  
jpeg                      9b                   h29c5dd3_2  
jsonschema                2.6.0            py36had8cd67_0  
jupyter                   1.0.0            py36h09218d1_0  
jupyter_client            5.1.0            py36hb51a33d_0  
jupyter_console           5.2.0            py36h4e8b69f_1  
jupyter_core              4.4.0            py36h2e6559f_0  
jupyterlab                0.30.6           py36h2f9c1c0_0  
jupyterlab_launcher       0.6.0                    py36_0  
lazy-object-proxy         1.3.1            py36h644e0db_0  
libedit                   3.1                  h9ac1fc1_0  
libffi                    3.2.1                h97ff0df_4  
libgcc-ng                 7.2.0                h7cc24e2_2  
libgfortran-ng            7.2.0                h9f7466a_2  
libpng                    1.6.32               h55ee776_4  
libsodium                 1.0.15               h1d95d67_0  
libssh2                   1.8.0                hcf52db8_3  
libstdcxx-ng              7.2.0                h7a57d05_2  
libtiff                   4.0.9                hb474d08_0  
libtool                   2.4.6                h7f8a5a3_3  
libxcb                    1.12                 h2ed087e_4  
libxml2                   2.9.4                h078cfa5_6  
libxslt                   1.1.29               h1a1cf83_6  
llvmlite                  0.21.0           py36h0fc0c68_0  
locket                    0.2.0            py36h930f432_1  
lxml                      4.1.1            py36h1054c7d_0  
lzo                       2.10                 ha1233cc_2  
markupsafe                1.0              py36h71540ef_1  
matplotlib                2.1.1            py36ha26af80_0  
mccabe                    0.6.1            py36h5a5e589_0  
mistune                   0.8.1            py36he5af1e3_0  
mkl                       2018.0.1             h4fb54b9_4  
mkl-service               1.1.2            py36hcf4eead_4  
mpc                       1.0.3                h760956b_5  
mpfr                      3.1.5                h2a7e9f2_2  
mpmath                    1.0.0            py36h1fc4c27_2  
msgpack-python            0.4.8            py36h8ae351a_0  
multipledispatch          0.4.9            py36h23d31ae_0  
navigator-updater         0.1.0            py36h54dd446_0  
nbconvert                 5.3.1            py36h4e2459c_0  
nbformat                  4.4.0            py36h2288295_0  
ncurses                   6.0                  h8a800b7_2  
networkx                  2.0              py36h642c725_0  
nltk                      3.2.5            py36h77327d7_0  
nose                      1.3.7            py36hd32e7dc_2  
notebook                  5.2.2            py36h6c9dd30_0  
numba                     0.36.1          np113py36h60023d9_0  
numexpr                   2.6.4            py36hcdb4828_0  
numpy                     1.13.3           py36hde72de1_0  
numpydoc                  0.7.0            py36h0fde76f_0  
odo                       0.5.1            py36h1f18d10_0  
olefile                   0.44             py36hec4b767_0  
openpyxl                  2.4.9            py36h1f00d45_0  
openssl                   1.0.2n               hb7f436b_0  
packaging                 16.8             py36hd57df8d_0  
pandas                    0.21.1           py36h20b78c2_0  
pandoc                    1.15.0.6             h57d12dc_1  
pandocfilters             1.4.2            py36h1ad6db2_1  
pango                     1.40.11              h523ddf7_0  
parso                     0.1.1            py36hf7c7001_0  
partd                     0.3.8            py36hf030253_0  
patchelf                  0.9                  hd2f9bfa_2  
path.py                   10.5             py36h29f8809_0  
pathlib2                  2.3.0            py36ha4cc02c_0  
patsy                     0.4.1            py36h70e543c_0  
pcre                      8.41                 he509c7b_1  
pep8                      1.7.0            py36h6aff9d3_0  
pexpect                   4.3.0            py36h854b63c_0  
pickleshare               0.7.4            py36hef8e16a_0  
pillow                    4.3.0            py36h24e0eac_1  
pip                       9.0.1            py36hdf7ad74_4  
pixman                    0.34.0               h8c3645b_3  
pkginfo                   1.4.1            py36h2f06541_0  
pluggy                    0.6.0            py36hade4285_0  
ply                       3.10             py36he66adc8_0  
prompt_toolkit            1.0.15           py36h1d3e51a_0  
psutil                    5.4.1            py36habb34c0_0  
ptyprocess                0.5.2            py36ha382e3e_0  
py                        1.5.2            py36h73b8c33_0  
pycodestyle               2.3.1            py36h102fe69_0  
pycosat                   0.6.3            py36h899c7ef_0  
pycparser                 2.18             py36h0ba8f0e_1  
pycrypto                  2.6.1            py36heae77cb_1  
pycurl                    7.43.0           py36h4e95443_3  
pyflakes                  1.6.0            py36h3256277_0  
pygments                  2.2.0            py36h129e8b3_0  
pylint                    1.7.4            py36he034cb8_0  
pympler                   0.5              py36h5bfab86_0  
pyodbc                    4.0.21           py36hd755fb1_0  
pyopenssl                 17.5.0           py36h54ad7e4_0  
pyparsing                 2.2.0            py36haacafa0_0  
pyqt                      5.6.0            py36h656727b_5  
pysocks                   1.6.7            py36hd51c7f5_1  
pytables                  3.3.0               np113py36_0  
pytest                    3.3.0            py36h146da80_0  
python                    3.6.3                hff703c5_8  
python-dateutil           2.6.1            py36h0ec02a2_1  
pytz                      2017.3           py36he1f0df5_0  
pywavelets                0.5.2            py36ha460e8f_0  
pyyaml                    3.12             py36ha50281b_1  
pyzmq                     16.0.3           py36h8b38b31_0  
qt                        5.6.2               h6a616b1_12  
qtawesome                 0.4.4            py36h799d091_0  
qtconsole                 4.3.1            py36h59098de_0  
qtpy                      1.3.1            py36hf9b8db7_0  
readline                  7.0                  h6b4b497_4  
requests                  2.18.4           py36hd88bebb_1  
rope                      0.10.7           py36h0f53b4d_0  
ruamel_yaml               0.11.14          py36h321a9fe_2  
scikit-image              0.13.0           py36h34637e1_1  
scikit-learn              0.19.1           py36h4fdfdbb_0  
scipy                     1.0.0            py36h5a0cb3a_0  
seaborn                   0.8.1            py36h50c05f2_0  
setuptools                36.5.0           py36h9e9eae1_0  
simplegeneric             0.8.1            py36h241e101_0  
singledispatch            3.4.0.3          py36he3f83a0_0  
sip                       4.18.1           py36h975b940_2  
six                       1.11.0           py36hd5049a8_1  
snowballstemmer           1.2.1            py36h4d4c73d_0  
sortedcollections         0.5.3            py36h1a518cb_0  
sortedcontainers          1.5.7            py36hbec566b_0  
sphinx                    1.6.3            py36hce11a15_0  
sphinxcontrib             1.0              py36ha84465b_1  
sphinxcontrib-websupport  1.0.1            py36hbf20963_1  
spyder                    3.2.4            py36h94cf2e7_0  
sqlalchemy                1.1.13           py36hfade055_0  
sqlite                    3.20.1               h27bc12f_2  
statsmodels               0.8.0            py36h55f7f34_0  
sympy                     1.1.1            py36h0c660c0_0  
tblib                     1.3.2            py36h26a6bb7_0  
terminado                 0.6              py36h7c16707_0  
testpath                  0.3.1            py36h69249d3_0  
tk                        8.6.7                h9a736a7_3  
toolz                     0.8.2            py36h3ec5cdc_0  
tornado                   4.5.2            py36hb54270b_0  
traitlets                 4.3.2            py36h6ad27bd_0  
typing                    3.6.2            py36h4e51af0_0  
unicodecsv                0.14.1           py36h59fad81_0  
unixodbc                  2.3.4                h090bd3b_1  
urllib3                   1.22             py36h0291d63_0  
wcwidth                   0.1.7            py36h9c8979c_0  
webencodings              0.5.1            py36ha860499_1  
werkzeug                  0.12.2           py36h68ce136_0  
wheel                     0.30.0           py36h89e8202_1  
widgetsnbextension        3.0.8            py36h1b5d3ec_0  
wrapt                     1.10.11          py36hbfa4134_0  
xlrd                      1.1.0            py36hbfbf212_1  
xlsxwriter                1.0.2            py36hb8a8e5b_0  
xlwt                      1.3.0            py36h75d342b_0  
xz                        5.2.3                ha3e1f4e_2  
yaml                      0.1.7                h69be6c5_2  
zeromq                    4.2.2                hcafe592_2  
zict                      0.1.3            py36habb5772_0  
zlib                      1.2.11               h907e355_2  
zope                      1.0                      py36_0  
zope.interface            4.4.3            py36hdae7dd0_0 

#### Expected Output
Loading a csv file to a DataFrame object

**Comment by jreback** [comment]
you would have to show how what you are doing is different. Is there a reason you are using 32-bit at all? 

pls show ``pd.show_versions()``
```
In [4]: !cat test.csv
a,b,c
1,2,3
4,5,6

In [5]: !ls -ltr *.csv
-rw-r--r--  1 jreback  staff  18 Dec 23 16:39 test.csv

In [6]: !pwd
/Users/jreback/pandas

In [7]: pd.read_csv('/Users/jreback/pandas/test.csv')
Out[7]: 
   a  b  c
0  1  2  3
1  4  5  6
```

**Comment by mknorps** [comment]
I have to use 32 bit system, because I am working on a virtual machine, and my DOS does not allow 64-bit linux installation. I cannot change it.

Output of pd.show_versions():
```
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.3.final.0
python-bits: 32
OS: Linux
OS-release: 4.10.0-42-generic
machine: i686
processor: i686
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.21.1
pytest: 3.3.0
pip: 9.0.1
setuptools: 36.5.0.post20170920
Cython: 0.27.3
numpy: 1.13.3
scipy: 1.0.0
pyarrow: None
xarray: None
IPython: 6.2.1
sphinx: 1.6.3
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.3
blosc: None
bottleneck: 1.2.1
tables: 3.3.0
numexpr: 2.6.4
feather: None
matplotlib: 2.1.1
openpyxl: 2.4.9
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.2
lxml: 4.1.1
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.1.13
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
```

**Comment by krassowski** [comment]
@mknorps, could you please run the following piece of code on your machine and see if it throws "Illegal instruction" too (that's the case for me)?

```
import numpy as np

a = np.arange(0, 100, 1)
np.where(a>90)
```
If yes, then we could narrow down the issue to https://github.com/numpy/numpy/issues/9532 and other issues related to AVX2 support detection. If not, what do you see if you run:

```gdb -ex r --args python project.py```

(you will need to install gdb first)

**Comment by mknorps** [comment]
@krassowski bingo!
I got the same issue with np.where(a>90).
I think that this issue is solved in the newest version of numpy.

With following setup all works fine:
>>> pd.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.3.final.0
python-bits: 32
OS: Linux
OS-release: 4.10.0-42-generic
machine: i686
processor: i686
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.22.0
pytest: None
pip: 9.0.1
setuptools: 38.4.0
Cython: None
numpy: 1.14.0
scipy: None
pyarrow: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.6.1
pytz: 2017.3
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

**Comment by mknorps** [comment]
I thought this issue is over, but when I wanted to contribute and fix one of pandas small issues the same 
'''
Illegal instruction (core dumped)
'''
are back.

My current package setup:

'''
>>> pd.show_versions()

INSTALLED VERSIONS
------------------
commit: 45aa0ce963df5c766063d7f4bfc6493b852028ba
python: 3.6.3.final.0
python-bits: 32
OS: Linux
OS-release: 4.10.0-42-generic
machine: i686
processor: i686
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.23.0.dev0+155.g45aa0ce.dirty
pytest: 3.3.2
pip: 9.0.1
setuptools: 28.8.0
Cython: 0.27.3
numpy: 1.14.0
scipy: None
pyarrow: None
xarray: None
IPython: None
sphinx: 1.6.6
patsy: None
dateutil: 2.6.1
pytz: 2017.3
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
'''

**Comment by mknorps** [comment]
Unfortunately still not fine. When I run the same pandas & numpy setup on virtual machine with Python 2.5.2 the error appeared again.
I created a test file test.py with code from @krassowski and debugging gives me:
```
(venv) knorpma1@knorpma1-VirtualBox:~/Documents/DATA_projects$ gdb -ex r --args python test.py
GNU gdb (Ubuntu 7.11.1-0ubuntu1~16.5) 7.11.1
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "i686-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from python...(no debugging symbols found)...done.
Starting program: /home/knorpma1/Documents/DATA_projects/venv/bin/python test.py
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/i386-linux-gnu/libthread_db.so.1".

Program received signal SIGILL, Illegal instruction.
0xb5f87218 in LONG_greater_avx2 (args=args@entry=0xbfffe54c, dimensions=dimensions@entry=0xbfffe558, steps=steps@entry=0xbfffe564, 
    __NPY_UNUSED_TAGGEDfunc=__NPY_UNUSED_TAGGEDfunc@entry=0x0) at numpy/core/src/umath/loops.c.src:936
936	numpy/core/src/umath/loops.c.src: No such file or directory.

```

**Comment by mknorps** [comment]
solved in [numpy/numpy#9532](https://github.com/numpy/numpy/issues/9532)
